### PR TITLE
[2.9] Clean up orphaned ClusterUserAttribute

### PR DESCRIPTION
**Pull request by pmatseykanets**
Ref: https://github.com/rancher/rancher/issues/44914

## Problem

If the ClusterUserAttribute becomes orphaned the corresponding controller will spin not being able to find related UserAttribute.
 
## Solution

Delete orphaned ClusterUserAttribute's. 
 
### Automated Testing

* Test types added/modified:
    * Unit

### Regressions Considerations

N/A

Existing / newly added automated tests that provide evidence there are no regressions:
N/A

### CODE REVIEW

**Review by crobby — DISMISSED** [review]
lgtm

**Review by MbolotSuse — CHANGES_REQUESTED** [review]
A couple of comments, and one issue that I think is a bug.

File: pkg/controllers/managementuser/clusterauthtoken/cluster_user_attribute.go [hunk 1]
```diff
@@ -17,15 +22,31 @@ func (h *clusterUserAttributeHandler) Sync(key string, clusterUserAttribute *clu
...
-		return nil, nil
-	}
-
 	userAttribute, err := h.userAttributeLister.Get("", clusterUserAttribute.Name)
 	if err != nil {
-		return nil, err
+		// If the corresponding UserAttribute no longer exists, we need to clean up the ClusterUserAttribute.
+		if apierrors.IsNotFound(err) {
+			// There is a chance that it hasn't ended up in the cache yet so we try to get it afresh.
+			userAttribute, err = h.userAttribute.Get("", metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (pmatseykanets) [reply pkg/controllers/managementuser/clusterauthtoken/cluster_user_attribute.go]: Duh 🤦‍♂️ . Thanks for catching this!

File: pkg/controllers/managementuser/clusterauthtoken/cluster_user_attribute.go [hunk 2]
```diff
@@ -17,15 +22,31 @@ func (h *clusterUserAttributeHandler) Sync(key string, clusterUserAttribute *clu
...
+				}
+				return nil, err
+			}
+			// The userAttribute exists, proceed.
+		} else {
+			return nil, err
+		}
 	}
 
+	if !clusterUserAttribute.NeedsRefresh {
```
[the remark this answers is not in the document]
Comment (pmatseykanets) [reply pkg/controllers/managementuser/clusterauthtoken/cluster_user_attribute.go:47]: We can leave it where it was L20 but then we'll be checking for missing user attribute  only when  `clusterUserAttribute.NeedsRefresh == true`.

[the remark this answers is not in the document]
Comment (MbolotSuse) [reply pkg/controllers/managementuser/clusterauthtoken/cluster_user_attribute.go:47]: Resolving, while this isn't strictly necessary leaving it behind the `NeedsRefresh` gate could also have some bad impacts.

**Review by MbolotSuse — APPROVED** [review]

**Comment by MbolotSuse** [comment]
@pmatseykanets LGTM, but please squash the commits when merging.

**Comment by pmatseykanets** [comment]
> but please squash the commits when merging.
Always

**Review by JonCrowther — DISMISSED** [review]

**Review by crobby — APPROVED** [review]

**Pull request merged**